packages/okta-react-native/ios/OktaSdkBridge.swift

```diff
@@ -146,14 +146,18 @@ class OktaSdkBridge: RCTEventEmitter {
     }
 
     @objc
-    func authenticate(_ sessionToken: String) {
+    func authenticate(_ sessionToken: String,
+                      promiseResolver: @escaping RCTPromiseResolveBlock,
+                      promiseRejecter: @escaping RCTPromiseRejectBlock) {
         guard let _ = config, let currOktaOidc = oktaOidc else {
             let error = OktaReactNativeError.notConfigured
             let errorDic = [
                 OktaSdkConstant.ERROR_CODE_KEY: error.errorCode,
                 OktaSdkConstant.ERROR_MSG_KEY: error.errorDescription
             ]
             sendEvent(withName: OktaSdkConstant.ON_ERROR, body: errorDic)
```

Comment:
If you introduce breaking change with callbacks then do we really need to leave events approach?

Reply:
Events can still kept to make sure it's backwards compatible. Also, the events approach would still be needed for the OIDC flow.

Reply:
Ok, if it is not a breaking change then we definitely need to keep it